### PR TITLE
signup feature implementation

### PR DESCRIPTION
**Pull request by akhaled7952023**
Implemented Features:

Designed and implemented the signup form with validation.
Integrated the API for user signup.
Successfully handling API responses for success scenarios.
Pending Issues:

Error Handling: While the API correctly returns errors, currently the app only displays success messages in the Awesome Dialog. Error cases are not yet displayed appropriately.
Localization: Work on supporting multiple languages is still pending.

### CODE REVIEW

File: lib/src/presentation/auth/signup/views/SignUpModelViewListener.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<SignUpViewModel, SignupViewState>(
+      listener: (context, state) {
+        if (state is SignupLoadingState) {
+        const  CircularProgressIndicator();
+        } else if (state is SignupSuccessState) {
+          showAwesomeDialog(
+            context,
+            title: "Success",
```
Comment (ali72-20) [inline lib/src/presentation/auth/signup/views/SignUpModelViewListener.dart:22]: use local avoid hard coded string

File: lib/src/presentation/splash.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            Text(
+              'Welcome to Flower App',
+              style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
+            ),
+            SizedBox(height: 20),
+            ElevatedButton(
+              onPressed: () {
+                Navigator.pushNamed(context, PageRouteName.signup);
+              },
+              child: Text('Go to Sign Up'),
```
Comment (ali72-20) [inline lib/src/presentation/splash.dart:24]: avoid hard coded string

**Review by ali72-20 — APPROVED** [review]
done

**Review by OmarHamed0 — APPROVED** [review]

File: lib/src/presentation/auth/signup/views/SignUpModelViewListener.dart (hunk 1 above)
Comment (akhaled7952023) [reply lib/src/presentation/auth/signup/views/SignUpModelViewListener.dart:22]: Ok , i'll done it

**Pull request merged**